#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 7b046fca3b145bc49ab3d59bf428b93b1dbc4e00 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Wed, 22 Nov 2023 18:44:31 +0100
Subject: [PATCH] Revert "Change threading / blocking when loading aboutdata"

This reverts commit d7808bb696426eb000cab74f69a0babc13e313be.

This change somehow caused the updateAboutDataFromSettings
to be ignored and only added the backend versions to
the default about data. This needs a better fix or better
understanding.
---
 src/aboutdata.cpp | 64 +++++++++++++++++++++++------------------------
 1 file changed, 32 insertions(+), 32 deletions(-)

diff --git a/src/aboutdata.cpp b/src/aboutdata.cpp
index f5c72174b..764fa6a82 100644
--- a/src/aboutdata.cpp
+++ b/src/aboutdata.cpp
@@ -85,44 +85,44 @@ void updateAboutDataFromSettings(const QSettings *settings)
 // make a big difference with regards to the available features.
 static void loadBackendVersions()
 {
-    auto thread = QThread::create([]() {
-        STARTUP_TIMING << "Checking backend versions";
-        const auto backendVersions = Kleo::backendVersionInfo();
-        STARTUP_TIMING << "backend versions checked";
-        if (!backendVersions.empty()) {
-            QMetaObject::invokeMethod(qApp, [backendVersions]() {
-                auto about = KAboutData::applicationData();
-                about.setOtherText(i18nc("Preceeds a list of applications/libraries used by Kleopatra", "Uses:") //
-                                   + QLatin1String{"<ul><li>"} //
-                                   + backendVersions.join(QLatin1String{"</li><li>"}) //
-                                   + QLatin1String{"</li></ul>"} //
-                                   + about.otherText());
-                KAboutData::setApplicationData(about);
-            });
-        }
-    });
-    thread->start();
+    STARTUP_TIMING << "Checking backend versions";
+    const auto backendVersions = Kleo::backendVersionInfo();
+    STARTUP_TIMING << "backend versions checked";
+    if (!backendVersions.empty()) {
+        auto about = KAboutData::applicationData();
+        about.setOtherText(i18nc("Preceeds a list of applications/libraries used by Kleopatra", "Uses:") //
+                           + QLatin1String{"<ul><li>"} //
+                           + backendVersions.join(QLatin1String{"</li><li>"}) //
+                           + QLatin1String{"</li></ul>"} //
+                           + about.otherText());
+        KAboutData::setApplicationData(about);
+    }
 }

 // This code is mostly for Gpg4win and GnuPG VS-Desktop so that they
 // can put in their own about data information.
 static void loadCustomAboutData()
 {
-    const QStringList searchPaths = {Kleo::gnupgInstallPath()};
-    const QString versionFile = QCoreApplication::applicationDirPath() + QStringLiteral(VERSION_RELPATH);
-    const QString distSigKeys = Kleo::gnupgInstallPath() + QStringLiteral(GNUPG_DISTSIGKEY_RELPATH);
-    STARTUP_TIMING << "Starting version info check";
-    bool valid = Kleo::gpgvVerify(versionFile, QString(), distSigKeys, searchPaths);
-    STARTUP_TIMING << "Version info checked";
-    if (valid) {
-        qCDebug(KLEOPATRA_LOG) << "Found valid VERSION file. Updating about data.";
-        auto settings = std::make_shared<QSettings>(versionFile, QSettings::IniFormat);
-        settings->setIniCodec(QTextCodec::codecForName("UTF-8"));
-        settings->beginGroup(QStringLiteral("Kleopatra"));
-        updateAboutDataFromSettings(settings.get());
-        KleopatraApplication::instance()->setDistributionSettings(settings);
-    }
-    loadBackendVersions();
+    auto thread = QThread::create([]() {
+        const QStringList searchPaths = {Kleo::gnupgInstallPath()};
+        const QString versionFile = QCoreApplication::applicationDirPath() + QStringLiteral(VERSION_RELPATH);
+        const QString distSigKeys = Kleo::gnupgInstallPath() + QStringLiteral(GNUPG_DISTSIGKEY_RELPATH);
+        STARTUP_TIMING << "Starting version info check";
+        bool valid = Kleo::gpgvVerify(versionFile, QString(), distSigKeys, searchPaths);
+        STARTUP_TIMING << "Version info checked";
+        QMetaObject::invokeMethod(qApp, [versionFile, valid]() {
+            if (valid) {
+                qCDebug(KLEOPATRA_LOG) << "Found valid VERSION file. Updating about data.";
+                auto settings = std::make_shared<QSettings>(versionFile, QSettings::IniFormat);
+                settings->setIniCodec(QTextCodec::codecForName("UTF-8"));
+                settings->beginGroup(QStringLiteral("Kleopatra"));
+                updateAboutDataFromSettings(settings.get());
+                KleopatraApplication::instance()->setDistributionSettings(settings);
+            }
+            loadBackendVersions();
+        });
+    });
+    thread->start();
 }

 AboutData::AboutData()
--
2.42.1
